addon/webextension/background/selectorLoader.js

```diff
@@ -0,0 +1,74 @@
+/* globals browser, catcher */
+window.selectorLoader = (function () {
+  const exports = {};
+  // These modules are loaded in order, and some need to be listed before others
+  // due to dependencies:
+  const scripts = [
+    "catcher.js",
+    "background/selectorLoader.js",
+    "selector/callBackground.js",
+    "clipboard.js",
+    "makeUuid.js",
+    "build/shot.js",
+    "randomString.js",
+    "domainFromUrl.js",
+    "build/inlineSelectionCss.js",
+    "selector/documentMetadata.js",
+    "selector/util.js",
+    "selector/ui.js",
+    "selector/shooter.js",
+    "selector/uicontrol.js"
+  ];
+
+  exports.unloadIfLoaded = function () {
+    return browser.tabs.executeScript({
+      code: "window.selectorLoader && window.selectorLoader.unloadModules()"
+    }).then(result => {
+      return result && result.toString() === "true";
+    });
+  };
+
+  exports.loadModules = function () {
+    return scripts.reduce(
+      (previous, file) =>
+        previous.then(() => browser.tabs.executeScript({file})),
+      Promise.resolve(null))
+      .catch(err => {
+        console.error(`Error loading script: ${err}`);
+        catcher.unhandled(err);
+        exports.unloadModules();
+      });
+  };
```

Comment:
I liked it the previous way, reduce always seems too clever to me.  Also I believe the previous logic included the script name when there was an error, and when there's an error the script name is very helpful (and not always available).

Also I think unloadModules is a good guard when there's an error, but we should let the caller handle the error and not do catcher.unhandled here.